[SERV-774] Make POST endpoints accept arrays

markmatney opened the pull request:
Since c840f4a56ef30b259aa2ddc8ac1e2daf455b660e renames files, I'd say the clearest picture of the changes can be seen with:

```bash
git diff ad56..0603
```

Summary:
- AddInstitution and AddJob pluralized throughout codebase; some methods now take a List instead of a single object, and handle that change as appropriate in their implementations
- When rewriting the tests, I realized they would be more concise if Institution and Job could be directly compared for equality, rather than comparing their JSON representations (even though equality in either sense implies equality in both)

d0427f2814ef53d7e1a08f8fbc5115a426a8f1a5 and 88c1239fb4898e6921c1ffed84e6223dccb2a95a are unrelated / just polish.

Commit:
[SERV-774] Make POST endpoints accept arrays

Commit:
Rename handlers to plural form

Commit:
Implement equals, hashCode, toString for data classes

Commit:
Use more concise logging syntax

Commit:
Use SqlTemplate instead of PreparedQuery for consistency

ksclarke commented:
The `libs` dir shouldn't affect anything but the dependency that's pulled from it. I think the failed GA builds were a result of a temporary Maven Central resolution problem. I restarted the GA and it succeeds now.

I do notice a lot of errors in the logs, despite the build not being failed. That's intended? I know sometimes tests can result in misleading errors (because they're testing things that should throw errors).

markmatney commented:
Yeah, those errors are OK from what I can tell.

Commit:
Scope tighter a couple new helper functions

Commit:
Recant my utterances of the most cursed word

Commit:
Make method static that can be

Commit:
Fix missing pluralizations

Commit:
Remove assertions causing test brittleness

These make certain assumptions about the list of institutions which is
not guaranteed.

markmatney commented:
Ready for re-review. The last two commits make it a bad request to submit an empty list (which was an internal server error earlier).

DRickard commented on src/main/java/edu/ucla/library/prl/harvester/Institution.java:
```diff
@@ -352,4 +352,43 @@ public URL getWebsite() {
+        if (anOther instanceof Institution) {
+            final Institution other = (Institution) anOther;
+
+            if (getID().equals(other.getID()) && getName().equals(other.getName()) &&
```
Any opinion on trimming whitespace in the string fields, to eliminate typos that might produce a false negative? Or trimming strings before add/update operations? Would also apply to Job.java

markmatney replied:
Sure, I could see that happening during construction. I would prefer that as a separate PR though.

ksclarke replied:
Create a new ticket for it?

markmatney replied:
SERV-791

Commit:
Add tests that fail for empty lists

Commit:
Make tests pass
